src/Psalm/Internal/Analyzer/Statements/ExpressionAnalyzer.php

```diff
@@ -147,12 +147,27 @@ public static function analyze(
                     $stmt->inferredType = Type::getLiteralClassString($context->self);
                     break;
 
+                case '__namespace__':
+                    $namespace = $statements_analyzer->getNamespace();
+                    if ($namespace === null &&
+                    IssueBuffer::accepts(
+                        new UndefinedConstant(
```

Comment:
You can't be outside of a namespace, you're just in the global namespace.

Silently change this from null to the empty string, instead?

Reply:
I thought about this. The warning could be clearer, sure, but `__CLASS__` is also an empty string outside of classes and has a warning. I figured that the same reasoning applies.

The warning is helpful with `define()`. Try defining a constant with `define(__NAMESPACE__.'\\cons')` in the global namespace. There's no way to access it.